src/libraries/System.Private.Xml/tests/XmlSchema/XmlSchemaSet/TC_SchemaSet_Compile.cs

```diff
@@ -176,5 +176,41 @@ public void FractionDigitsMismatch_Throws()
             Assert.Contains("fractionDigits", ex.Message);
             Assert.DoesNotContain("totalDigits", ex.Message);
         }
+
+        [Fact]
+        public void TotalDigitsParseValue_Succeeds()
+        {
+            string schema = @"<?xml version='1.0' encoding='utf-8' ?>
+<xs:schema elementFormDefault='qualified'
+           xmlns:xs='http://www.w3.org/2001/XMLSchema'>
+    <xs:simpleType name='foo'>
+        <xs:restriction base='xs:decimal'>
+            <xs:totalDigits value='8' fixed='true'/>
+        </xs:restriction>
+    </xs:simpleType>
+    <xs:simpleType name='bar'>
+        <xs:restriction base='foo'>
+            <xs:totalDigits value='8'/>
+        </xs:restriction>
+    </xs:simpleType>
+</xs:schema>
+";
+            XmlSchemaSet ss = new XmlSchemaSet();
+            ss.Add(null, XmlReader.Create(new StringReader(schema)));
+
+            bool success;
+            try
+            {
+                ss.Compile();
+                success = true;
+            }
+            catch (Exception ex)
+            {
+                success = false;
+                Assert.IsNotType<InvalidCastException>(ex);
```

Comment:
I'm confused - what's this line for - the test will fail without it, right?

Reply:
@krwq 
That line was just in case it threw a different kind of Exception than the one I was testing for.  Per feedback above from @buyaa-n, the entire try/catch has been removed.